problem_builder/public/js/mentoring_with_steps.js

```diff
@@ -0,0 +1,157 @@
+function MentoringWithStepsBlock(runtime, element) {
+
+    var steps = runtime.children(element).filter(
+        function(c) { return c.element.className.indexOf('pb-mentoring-step') > -1; }
+    );
+    var step = $('.mentoring', element).data('step');
+    var active_child, checkmark, submitDOM, nextDOM, tryAgainDOM, submitXHR;
+
+    function isLastChild() {
+        return (active_child === steps.length-1);
+    }
+
+    function updateStep() {
+        var handlerUrl = runtime.handlerUrl(element, 'update_step');
+        $.post(handlerUrl, JSON.stringify(step+1))
+            .success(function(response) {
+                step = response.step;
```

Comment:
@itsjeyd nit: same applies here - step_number and update_step_number might be a better names for that.